ui/src/timeMachine/actions/queryBuilder.ts

```diff
@@ -10,220 +10,113 @@ import {
 
 // Types
 import {Dispatch} from 'redux-thunk'
-import {GetState} from 'src/types'
-import {RemoteDataState} from 'src/types'
+import {GetState, RemoteDataState} from 'src/types'
+import {BuilderAggregateFunctionType} from 'src/client/generatedRoutes'
 import {BuilderFunctionsType} from '@influxdata/influx'
 
 export type Action =
-  | SetBuilderBucketSelectionAction
-  | SetBuilderBucketsAction
-  | SetBuilderBucketsStatusAction
-  | SetBuilderTagKeysAction
-  | SetBuilderTagKeysStatusAction
-  | SetBuilderTagValuesAction
-  | SetBuilderTagValuesStatusAction
-  | SetBuilderTagKeySelectionAction
-  | SetBuilderTagValuesSelectionAction
-  | AddTagSelectorAction
-  | RemoveTagSelectorAction
-  | SetFunctionsAction
-  | SelectAggregateWindowAction
-  | SetValuesSearchTermAction
-  | SetKeysSearchTermAction
-  | SetBuilderTagsStatusAction
-
-interface SetBuilderBucketsStatusAction {
-  type: 'SET_BUILDER_BUCKETS_STATUS'
-  payload: {bucketsStatus: RemoteDataState}
-}
+  | ReturnType<typeof setBuilderAggregateFunctionType>
+  | ReturnType<typeof setBuilderBucket>
+  | ReturnType<typeof setBuilderBuckets>
+  | ReturnType<typeof setBuilderBucketsStatus>
+  | ReturnType<typeof setBuilderTagKeys>
+  | ReturnType<typeof setBuilderTagKeysStatus>
+  | ReturnType<typeof setBuilderTagValues>
+  | ReturnType<typeof setBuilderTagValuesStatus>
+  | ReturnType<typeof setBuilderTagKeySelection>
+  | ReturnType<typeof setBuilderTagValuesSelection>
+  | ReturnType<typeof addTagSelectorSync>
+  | ReturnType<typeof removeTagSelectorSync>
+  | ReturnType<typeof setFunctions>
+  | ReturnType<typeof selectAggregateWindow>
+  | ReturnType<typeof setValuesSearchTerm>
+  | ReturnType<typeof setKeysSearchTerm>
+  | ReturnType<typeof setBuilderTagsStatus>
+
+export const setBuilderAggregateFunctionType = (
+  builderAggregateFunctionType: BuilderAggregateFunctionType,
+  index: number
+) => ({
+  type: 'SET_BUILDER_AGGREGATE_FUNCTION_TYPE' as 'SET_BUILDER_AGGREGATE_FUNCTION_TYPE',
+  payload: {builderAggregateFunctionType, index},
+})
```

Comment:
apologies for having this diff include non-logical structural changes in this file (it's all in service of changing out actions to be less verbose)

this block is the main logical addition to this file